drivers/axi_core/axi_dmac/axi_dmac.c

```diff
@@ -47,6 +47,49 @@
 #include "delay.h"
 #include "axi_dmac.h"
 
+/***************************************************************************//**
+ * @brief dma_isr
+*******************************************************************************/
+void axi_dmac_isr(void *instance)
+{
+	struct axi_dmac *dmac = (struct axi_dmac *)instance;
+	uint32_t user_index = dmac->user_index, driver_index = dmac->driver_index;
+	uint32_t reg_temp;
+
+	axi_dmac_read(dmac, AXI_DMAC_REG_IRQ_PENDING, &reg_temp);
+	axi_dmac_write(dmac, AXI_DMAC_REG_IRQ_PENDING, reg_temp);
+
+	if (driver_index == user_index) {
+		dmac->usr_callback();
+		return;
+	}
+
+	switch (dmac->direction) {
+	case DMA_DEV_TO_MEM:
+		axi_dmac_write(dmac, AXI_DMAC_REG_DEST_ADDRESS,
+				dmac->transfer_lifo[driver_index].address);
+		axi_dmac_write(dmac, AXI_DMAC_REG_DEST_STRIDE, 0x0);
+		break;
+	case DMA_MEM_TO_DEV:
+		axi_dmac_write(dmac, AXI_DMAC_REG_SRC_ADDRESS,
+				dmac->transfer_lifo[driver_index].address);
+		axi_dmac_write(dmac, AXI_DMAC_REG_SRC_STRIDE, 0x0);
+		break;
+	default:
+		return; // Other directions are not supported yet
+	}
+	axi_dmac_write(dmac, AXI_DMAC_REG_X_LENGTH,
+			dmac->transfer_lifo[driver_index].size - 1);
+	axi_dmac_write(dmac, AXI_DMAC_REG_Y_LENGTH, 0x0);
+
+	axi_dmac_write(dmac, AXI_DMAC_REG_START_TRANSFER, 0x1);
+
+	if (dmac->driver_index < 10)
```

Comment:
nitpic: dmac->driver_index =  (dmac->driver_index + 1) % DMAC_LIFO_SIZE

Reply:
This has changed completely.